nixos/modules/services/networking/cloudflared.nix

```diff
@@ -0,0 +1,95 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.cloudflared;
+in
+{
+  meta.maintainers = with maintainers; [ bbigras ];
+
+  options.services.cloudflared = {
+    enable = mkEnableOption "Cloudflared Argo Tunnel client daemon";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.cloudflared;
+      defaultText = "pkgs.cloudflared";
+      description = "The package to use for Cloudflared";
+    };
+
+    tunnels = mkOption {
+      description = ''
+        Cloudflare tunnels
+      '';
+      type = types.attrsOf (types.submodule ({ name, ... }: {
+        options = {
+          tunnel = mkOption {
+            type = with types; nullOr str;
+            default = null;
+            description = ''
+              Tunnel ID
+            '';
+            example = "00000000-0000-0000-0000-000000000000";
+          };
+
+          credentialsFile = mkOption {
```

Comment:
I am not sure if this adds the file to store which would make it publicly readable.

Reply:
In my case I got:
```
"credentials-file":"/run/secrets/cloudflared-creds"
```

in the config file. So the config file is in the store but not the creds file.